src/render/context/mod.rs

```diff
@@ -220,6 +229,15 @@ impl Context {
     /// Initializes [Context], optionally reading in the configuration file to override defaults.
     /// Arguments provided will take precedence over config.
     pub fn init() -> Result<Self, Error> {
+        trait IntoVecOfStr {
```

Comment:
I understand that this is only used here, but I do find it strange to see trait declaration/impls being used inline like this. Could we move this to top-level of this module?